common/libs/wallet/deterministic/deterministic.ts

```diff
@@ -1,7 +1,7 @@
 export class DeterministicWallet {
-  private address: string;
-  private dPath: string;
-  private index: number;
```

Comment:
Is there any reason these need to be private? Declaring these as `private` means these variables were would be retrieved from the redux state then passed as arguments back into methods extending `DeterministicWallet`. Declaring them as `public` allows classes like `LedgerWallet` or `TrezorWallet`, which extend `DeterministicWallet`, can use `this.dPath, this.address, ...` in their methods instead of having them passed as arguments

Reply:
I believe what you'd want here is the `protected` keyword. Children of the class can access them, but not external code.

I'd prefer we keep as few attributes public as possible, and instead offer up functions wherever possible that do what need doing with the attributes. Just makes refactoring a bit easier.